cmd/build/v2/services_test.go

```diff
@@ -81,40 +45,94 @@ func TestServicesNotInStack(t *testing.T) {
-	ctx := context.Background()
-	toBuild, err := bc.GetServicesToBuild(ctx, fakeManifest, []string{"test-1"})
-	//should not throw error
-	assert.NoError(t, err)
-	assert.Equal(t, 1, len(toBuild))
-}
+func TestGetServicesToBuildWithManifestV2(t *testing.T) {
 
-func TestServicesNotAreAlreadyBuiltWithSubset(t *testing.T) {
-	fakeReg := test.NewFakeOktetoRegistry(nil)
-	bc := NewBuilder(nil, fakeReg)
-	alreadyBuilt := []string{"test/test-1"}
-	fakeReg.AddImageByName(alreadyBuilt...)
-	ctx := context.Background()
-	toBuild, err := bc.GetServicesToBuild(ctx, fakeManifest, []string{"test-1"})
-	//should not throw error
-	assert.NoError(t, err)
-	assert.Equal(t, 0, len(toBuild))
-}
+	testCases := []struct {
+		name                  string
+		manifestBuildImages   []string
+		alreadyBuilt          []string
+		servicesToDeploy      []string
+		expectedImagesToBuild []string
+	}{
+		{
+			name:                  "all services already built and deploy all",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1", "2", "3"},
+			servicesToDeploy:      []string{"1", "2", "3"},
+			expectedImagesToBuild: []string{},
+		},
+		{
+			name:                  "all services already built and deploy subset",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1", "2", "3"},
+			servicesToDeploy:      []string{"1", "2"},
+			expectedImagesToBuild: []string{},
+		},
+		{
+			name:                  "Same services to deploy and already built",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1", "2"},
+			servicesToDeploy:      []string{"1", "2"},
+			expectedImagesToBuild: []string{},
+		},
+		{
+			name:                  "some services already built",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1", "2"},
+			servicesToDeploy:      []string{"2", "3"},
+			expectedImagesToBuild: []string{"3"},
+		},
+		{
+			name:                  "No intesection between services to deploy and already built",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1"},
+			servicesToDeploy:      []string{"2", "3"},
+			expectedImagesToBuild: []string{"2", "3"},
+		},
+		{
+			name:                  "subset are already built",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1"},
+			servicesToDeploy:      []string{"1", "2"},
+			expectedImagesToBuild: []string{"2"},
+		},
+		{
+			name:                  "no services to deploy",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{"1"},
+			servicesToDeploy:      []string{},
+			expectedImagesToBuild: []string{},
+		},
+		{
+			name:                  "no services already built",
+			manifestBuildImages:   []string{"1", "2", "3"},
+			alreadyBuilt:          []string{},
+			servicesToDeploy:      []string{"1", "2"},
+			expectedImagesToBuild: []string{"1", "2"},
+		},
```

Comment:
could we face a situation where manifest (build section) doesn't have any of the servicesToDeploy? I mean, a test case like: `manifestBuildImages: A, B`and `servicesToDeploy: B, C`

Reply:
That sounds like a case we could have, for example by having a docker compose which some service comes from a public image. I'll think what the output to that should be and I'll add a test